lib/LoaderOptionsPlugin.js

```diff
@@ -2,33 +2,38 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var ModuleFilenameHelpers = require("./ModuleFilenameHelpers");
+"use strict";
 
-function LoaderOptionsPlugin(options) {
-	if(typeof options !== "object") options = {};
-	if(!options.test) options.test = {
-		test: function() {
-			return true;
-		}
-	};
-	this.options = options;
-}
-module.exports = LoaderOptionsPlugin;
+const ModuleFilenameHelpers = require("./ModuleFilenameHelpers");
 
-LoaderOptionsPlugin.prototype.apply = function(compiler) {
-	var options = this.options;
-	compiler.plugin("compilation", function(compilation) {
-		compilation.plugin("normal-module-loader", function(context, module) {
-			var resource = module.resource;
-			if(!resource) return;
-			var i = resource.indexOf("?");
-			if(ModuleFilenameHelpers.matchObject(options, i < 0 ? resource : resource.substr(0, i))) {
-				Object.keys(options).filter(function(key) {
-					return ["include", "exclude", "test"].indexOf(key) < 0
-				}).forEach(function(key) {
-					context[key] = options[key];
-				});
+class LoaderOptionsPlugin {
+	constructor(options) {
+		if(typeof options !== "object") options = {};
+		if(!options.test) options.test = {
+			test: function() {
+				return true;
 			}
+		};
+		this.options = options;
+	}
+
+	apply(compiler) {
+		const options = this.options;
+		compiler.plugin("compilation", (compilation) => {
+			compilation.plugin("normal-module-loader", (context, module) => {
+				const resource = module.resource;
+				if(!resource) return;
+				const i = resource.indexOf("?");
+				if(ModuleFilenameHelpers.matchObject(options, i < 0 ? resource : resource.substr(0, i))) {
+					Object.keys(options).filter((key) => {
+						return ["include", "exclude", "test"].indexOf(key) < 0
```

Comment:
This could be hoisted into a `const filterSet = new Set(["include", "exclude", "test"])` (next to the `const options`, so the filter can be simplified into `.filter(key => !filterSet.has(key))`.

I also remember having seen something in another PR about wanting `"object"`-typed values using `let` instead of `const`, but I don't understand the reasoning...